### DISCUSSION TITLE
How To Include Other JS Libs in Svelte-TS-Template?

### DISCUSSION THREAD

**Discussion by KiddoV** [wailsapp/wails#1970] [Q&A] [answered]
I just started using Svelte template with TS. It is awesome, and I love it.
However, I was moving my project from a completely plain JS to Svelte. In a "non-framework front-end" I would just include those libs in my HTML head tag like this:
```html
<head>
    <title>My App</title>
    <!-- JQuery (In General) -->
    <script src="./src/assets/libs/jquery.js"></script>
    <!-- Fomantic-UI -->
    <link href="./src/assets/libs/fomantic-ui/fomantic-ui.css" rel="stylesheet">
    <script src="./src/assets/libs/fomantic-ui/fomantic-ui.js"></script>
</head>
```
... in Svelte on the other hands, it would fail (404 file not found or jQuery is not defined...) when I build the application (not at dev tho).
Is there anyway that I can accomplish this in Svelte?
Thanks,

**Comment by misitebao** [comment]
Please search for the keyword "**svelte:head**", that might be what you are looking for.

https://svelte.dev/docs#template-syntax-svelte-head

**Reply by KiddoV** [reply]
This only tie to 1 component right? How do I make it globally so any ``*.svelte`` file can access the lib?

**Reply by leaanthony** [reply]
Normally you just import your 3rd party libraries using npm and include them. For standalone JS files, I would download them to a lib directory then just import them as local files.

3rd party lib example: https://splidejs.com/integration/svelte-splide/?ref=madewithsvelte.com
Local file: https://stackoverflow.com/questions/56839098/import-javascript-file-in-svelte

**Reply by KiddoV** [reply]
Non of the methods you provided are working, since the JS library I am using depends on jQuery lib. It works in ``dev``, but when build, it would throws ``jQuery`` is not defined in the auto generated ``index-abc123.js`` file.

**Reply by KiddoV** [reply]
@misitebao 
```html
<svelte:head>
  <script src="./assets/libs/jquery.js"></script>
  <script src="./assets/libs/fomantic-ui/fomantic-ui.js"></script>
</svelte:head>
```
I got this error:
```console
[vite-plugin-svelte] Error while preprocessing C:/.../frontend/src/app.svelte - C:/.../frontend/src/app.svelte:9:92: Unknown word
```

**Reply by KiddoV** [reply]
@leaanthony 
NPM is a great idea but this lib is still required to import dependencies in the head tag for some reason:
https://fomantic-ui.com/introduction/getting-started.html#installing-via-npm

> Once complete you should have a new directory dist/ which contains all the CSS and JavaScript files. You can now include these in your web page and start using Fomantic-UI.
> 
>  ```html
>  
>  <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.1/dist/jquery.min.js"></script>
>  <link rel="stylesheet" type="text/css" href="/dist/semantic.min.css">
>  <script src="/dist/semantic.min.js"></script>
>  ```

**Comment by KiddoV** [comment, marked as answer]
I finally I solved this issue by using ``import inject from '@rollup/plugin-inject';``
- vite.config.ts
```ts
import {defineConfig} from 'vite';
import {svelte} from '@sveltejs/vite-plugin-svelte';
import inject from '@rollup/plugin-inject';

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    inject({
      $: "jquery",
      jQuery: "jquery",
      "window.jQuery": "jquery"
  }),
  svelte()],
})
``` 
For everyone who still want to use UMD libs in a modern framework!!!!
